File: OOPS/oopsPython/abstraction.encapsulation/EncapsulationDemo.py
# ACCESS MODIFIERS
# __name - private
# _name - protected
# name - public

class Profile:

    def __init__(self):
        self.__name = "krishna"
        self.roll = 5

    def getName(self):
        return self.__name

    def setName(self, name):
        self.__name = name

isAuthorized = True

profile = Profile()
print(profile.roll)
# print(profile.__name)
print(profile.getName())

if isAuthorized:
    profile.setName("kavi")
    print("Name is updated - ", profile.getName())
else:
    print("You're unauthorized")

File: OOPS/oopsJava/abstraction.encapsulation/EncapsulationDemo.java
package in.kpriet.training.oops;

class Profile {

    private String name;
    String roll;

    Profile() {
        this.name = "madhu";
    }  

    public String getName() {
        return name;
    }

    public String setName(String name) {
        return this.name = name;
    }
}


class EncapsulationDemo {

    public static void main(String[] args) {
        
        boolean isAuthorized = false;

        Profile profile = new Profile();
        // System.out.println(profile.name);

        System.out.println(profile.getName());

        if(isAuthorized) {
            profile.setName("ashu");
            System.out.println("Name is updated" + profile.getName());
        } else {
            System.out.println("You're unauthorized");
        }   
    }
}
